File: 228-summary-ranges/summary-ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> ans;
        int n = nums.size();
        if (n == 0) return ans;
        int start = nums[0];
        for(int i=1;i<=n; i++) {
            // now check till where the array is Continuous
            // if whole array is continuous or till where it is we have to add it in vector string
            if(i == n || nums[i] != nums[i-1] + 1) {
                if(start == nums[i-1]) {
                    // whole array is continous
                    ans.push_back(to_string(start));
                }
                else {
                    // if not
                    ans.push_back(to_string(start) + "->" + to_string(nums[i-1]));
                }
                if(i<n) start = nums[i];
            }
        } 

        return ans;
    }
};

File: 228-summary-ranges/summary-ranges.go
func summaryRanges(nums []int) []string {
    n := len(nums)
    var ans []string
    if n == 0 {return ans}
    start := nums[0]
    for i := 1; i <= n; i++ {
        if i == n || nums[i] != nums[i-1] + 1 { 
            if start == nums[i-1] {
                ans = append(ans, strconv.Itoa(start))
            } else {
                ans = append(ans, strconv.Itoa(start)+"->"+strconv.Itoa(nums[i-1]))
            } 
            if i < n {
                start = nums[i]
            }
        }
    }
    return ans
}